### ISSUE TITLE
UnsuportedOperationException: JsonNull

### ISSUE THREAD

**Issue by bourgouinadrien** [marian-m12l/studio#72]
Good evening,

  I was just ending a pack for my daughter (on windows) when I had an issue on transfering to the lunii.
Logs :
`2020-04-24 23:41:27,422 [WARN ] [s.w.s.LibraryService]: Pack to transfer is in archive format. Converting to binary format and storing in temporary file: C:\Users\Adrien\AppData\Local\Temp\Triforce-6470d48c-a0eb-4875-8101-1b95afb3ad96-v1.zip1591689213836808193.pack
2020-04-24 23:41:27,424 [WARN ] [s.w.s.LibraryService]: Reading archive format pack
Exception in thread "Timer-4" java.lang.UnsupportedOperationException: JsonNull
        at com.google.gson.JsonElement.getAsString(JsonElement.java:192)
        at studio.core.v1.reader.archive.ArchiveStoryPackReader.read(ArchiveStoryPackReader.java:170)
        at studio.webui.service.LibraryService.getBinaryPackFile(LibraryService.java:127)
        at studio.webui.api.DeviceController$1.run(DeviceController.java:66)
        at java.base/java.util.TimerThread.mainLoop(Timer.java:556)
        at java.base/java.util.TimerThread.run(Timer.java:506)`

Maybe not linked but I am unable to set my official.json file because I have this return from lunii API when I give token : `< HTTP/1.1 401 Unauthorized
< Server: nginx
< Date: Fri, 24 Apr 2020 21:56:17 GMT
< Content-Type: application/json; charset=utf-8
< Content-Length: 48
< Connection: keep-alive
< Access-Control-Allow-Origin: *
< Cache-Control: no-cache
< Strict-Transport-Security: max-age=31556926; includeSubDomains; preload
<
{
  "error" : "Unable to validate signature."
}
`

This with the token I have in "C:\Users\USER\AppData\Roaming\Luniitheque\.local.properties"
And the file is like : 

```
#Fri Apr 24 23:53:07 CEST 2020
token={"firebase"\:"TOKEN"}
remember_me=0
```
Not the same thing I can see in your README (tokens.access_tokens.data.firebase)

![image](https://user-images.githubusercontent.com/18287157/80260278-cf02de00-8687-11ea-82ac-bd83ff46bcce.png)

[Triforce-6470d48c-a0eb-4875-8101-1b95afb3ad96-v1.zip](https://github.com/marian-m12l/studio/files/4531748/Triforce-6470d48c-a0eb-4875-8101-1b95afb3ad96-v1.zip)

Can you help me with that please ?
Thanks a lot !

**Comment by wistof** [comment]
Hi,

Your Pack seems uncompleted.
Use the 'verify diagram' feature, on the right bar, last icon.

![image](https://user-images.githubusercontent.com/13451357/80274223-37c67680-86d9-11ea-8b35-2440dfe36530.png)

**Comment by bourgouinadrien** [comment]
I already did but it seems that each time some links disappears
I'll retry but I am confident on what I did.

**Comment by bourgouinadrien** [comment]
Proof of unrecording what I update :
https://webm.red/view/Vdyr.webm

**Comment by wistof** [comment]
> Proof of unrecording what I update :
> https://webm.red/view/Vdyr.webm

Just to tell, I've test your pack with build 0.1.14-Snapshot (under Windows) and when I record it then open it, all the links stay. I've got only on error with the 'Hirule Stories' node which isn't set as start node.
Perhaps you may change this node by a 'Cover Node'.

**Comment by bourgouinadrien** [comment]
@wistof thanks again.
We are on two different version for windows, you are using dev version and for me stable version.
But it is possible that that bug is corrected in next version.
I shall try to compile source to get it running on 0.1.14 :)
For Hirule Stories node, start node, I made like for the example of a several instories action of a lunii pack.

I'll try to use 0.1.14 tomorrow.

**Comment by MiX-FR** [comment]
> @wistof thanks for your answer.
> I already did but it seems that each time some links disappears
> I'll retry but I am confident on what I did.
> 
> Edit: it seems that when I save my story, it doesn't save every transition between node.
> When I reopen it, they are still missing and start node also is in error

I have this issue and have reported an issue : https://github.com/marian-m12l/studio/issues/71

**Comment by bourgouinadrien** [comment]
Ok technically we are tomorrow from yesterday...
I installed maven (didn't run that in like 7 years...) and run it with 0.1.14.SNAPSHOT version.

I have the same stack exception when uploading to lunii.
But according to "Les histoires de Suzanne et Gaston", I made the right thing
![image](https://user-images.githubusercontent.com/18287157/80292137-35edc900-8754-11ea-9149-827f3c4ba049.png)
And there is an "issue" on original pack from Lunii.

So I don't know how to handle that :/

Seems, like I told in #71 that you can't mix nodes.
There is two groups separated by vertical line, you can use one or the other, not both.

**Comment by marian-m12l** [comment]
Hi, and thanks for the feedback.
The stacktrace tells me there's an option in an action node that links to nothing. Did you try with a simpler story pack?
I cannot watch your webm so I don't see how you are "loosing" some transitions. What's your workflow?
The so-called "simplified" and "expert" nodes are meant to be compatible with each other, but there may be some bugs.
Also, I wouldn't recommend running 0.1.14-SNAPSHOT as of now, as you'll get nothing worthy except for a bug when saving/exporting certain story packs...

**Comment by bourgouinadrien** [comment]
Hi @marian-m12l,
That's weird that you can't read webm but steps are :
Open the diagram in editor
Check diagram, there is errors. 
Correct errors, check diagram, all is OK. 
Save diagram, return to story list view, reopen diagram in edit mode.
Check diagram, errors are back. 

I'll try on something smaller.

**Comment by marian-m12l** [comment]
Will have to take a deeper look to your pack. Thanks for your feedback!

**Comment by marian-m12l** [comment]
There was indeed an embarrassing bug when using "backwards" links (created from to destination). Now fixed with release 0.1.14

**Comment by bourgouinadrien** [comment]
@marian-m12l works like a charm ! 
Thanks a lot dude :)
